src/meta/Dimensions.ts

```diff
@@ -0,0 +1,64 @@
+import { WidgetMeta, WidgetMetaProperties } from '../WidgetBase';
+import Map from '@dojo/shim/Map';
+
+export interface DimensionResults {
+	bottom: number;
+	height: number;
+	left: number;
+	right: number ;
+	scrollHeight: number;
+	scrollLeft: number;
+	scrollTop: number;
+	scrollWidth: number;
+	top: number;
+	width: number;
+}
+
+export default class Dimensions implements WidgetMeta {
+	private _nodes: Map<string, any>;
+
+	readonly requiresRender = true;
+
+	constructor(properties: WidgetMetaProperties) {
+		this._nodes = properties.nodes;
+	}
+
+	has(key: string): boolean {
+		return !!this._nodes.get(key);
+	}
+
+	get(key: string): DimensionResults {
```

Comment:
I think `DimensionResults` should probably be either `Readonly<DimensionResults>` or the interface itself have  readonly properties.